### PR TITLE
COMMERCE-11650 

### PR DESCRIPTION
**Pull request by jaydawu**
https://issues.liferay.com/browse/COMMERCE-11650
Resent from https://github.com/liferay-commerce/liferay-portal/pull/3849

### CODE REVIEW

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by brianikim** [comment]
Source Format Failure not related to my changes.

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductDetails.testcase [hunk 1]
```diff
@@ -1546,6 +1546,162 @@ definition {
...
+		}
+
+		task ("And a product Brake Pads' Low Stock Threshold is set to 1 and Allow Back Orders is enabled") {
+			CommerceJSONProductsAPI._patchCommerceProductConfiguration(
+				allowBackOrder = "true",
+				productName = "Brake Pads");
+
+			CommerceProducts.openProductsAdmin();
+
+			CommerceNavigator.searchEntry(entryName = '''"Brake Pads"''');
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductDetails.testcase:1608]: You can use this macro to search and navigate to product entry:

CommerceProducts.searchAndGoToProductEntry(entryProductName = "Brake Pads");

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductDetails.testcase [hunk 2]
```diff
@@ -1546,6 +1546,162 @@ definition {
 		}
 	}
 
+	@description = "This is a test for COMMERCE-11650. User can checkout a product in a channel that has also stock in another channel, and the product's stock in another channel cannot be affected by this channel"
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductDetails.testcase:1549]: "As a user, when I place an order with a product that has also stock in a second channel, then the product's stock in the second channel won't be affected by the order made in the first one."

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductDetails.testcase [hunk 3]
```diff
@@ -1546,6 +1546,162 @@ definition {
...
+				locator1 = "AppBuilder#CHOOSE_APP_SEARCH_FILTER_BY_OPTION");
+
+			Navigator.openWithAppendToBaseURL(urlAppend = "web/minium-2/p/brake-pads");
+
+			AssertElementPresent(
+				key_filterOption = "6 in Stock",
+				locator1 = "AppBuilder#CHOOSE_APP_SEARCH_FILTER_BY_OPTION");
+		}
+
+		task ("When the buyer checkout with 11 quantities of Brake Pads on Minium") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductDetails.testcase:1659]: When the buyer places an order with 11 quantities of Brake Pads on Minium

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductDetails.testcase [hunk 4]
```diff
@@ -1546,6 +1546,162 @@ definition {
...
+			CommerceAccelerators.initializeNewSiteViaAccelerator(
+				siteName = "Minium 2",
+				siteTemplateName = "Minium");
+
+			JSONUser.addUserToSite(
+				groupName = "Minium 2",
+				userEmailAddress = "buyer@liferay.com");
+		}
+
+		task ("And an Italy warehouse eligible for Minium Portal and a Southwest warehouse eligible for Minium 2 Portal are set") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductDetails.testcase:1581]: I guess you are setting eligibility for Northeast for Minium 2 but wrote Southwest here:

And the Italy warehouse is set eligible for Minium Portal while the Northeast warehouse is set eligible for Minium 2 Portal

**Review by baharturk — CHANGES_REQUESTED** [review]
Hi, please take a look at my comments. Thanks!

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
@baharturk Thank you so much for the review. I updated the test, please take a look again, Thanks!

**Review by baharturk — APPROVED** [review]

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
Review passed, waiting for the CI results, then it can be forwarded or force-forwarded, thanks.

cc/ @victoroscuro

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
@victoroscuro the errors in relevant are not related to this pull, it can be force-forwarded, thank you!

**Comment by victoroscuro** [comment]
ci:forward:force

**Pull request closed without merging**